### PR TITLE
dev,3.0: Add backup and restore documentation

### PR DESCRIPTION
**Pull request by aylei**
### What is changed, added or deleted? 
Add backup and restore documentation.

CN: https://github.com/pingcap/docs-cn/pull/1651

### Which version does your change affect? 
`3.0`,`dev`

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

**Comment by aylei** [comment]
@tennix @DanielZhangQD @lilin90 @tomdewan PTAL

**Comment by tomdewan** [comment]
Thanks for the heads up. I will look at this tomorrow, 7/23 US times.

Thanks,

Tom

**Comment by aylei** [comment]
@tomdewan Sorry for disturbing, I mean to @ @TomShawn but made a typo😅

**Comment by tomdewan** [comment]
That's OK. :-) Thanks for letting me know.

Tom

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+If you store your backup data to [Google Cloud Storage](https://cloud.google.com/storage/), [Ceph Object Storage](https://ceph.com/ceph-storage/object-storage/) or [Amazon S3](https://aws.amazon.com/s3/), you can view the backups by using the GUI or CLI tool of these storage provider.
+
+## Restore
+
+The helm chart `pingcap/tidb-backup` helps restore a TiDB cluster using backup data. Refer to the steps below to perform a restore operation:
+
+* Modify the `values.yaml` file:
+    * Set `clusterName` to the target TiDB cluster name.
+    * Set `mode` to `restore`.
+    * Set `name` to the backup name you want to restore([view backups](#view-backups) helps you view all the backups available). If the backup is stored in `Google Cloud Storage`, `Ceph Object Storage` or `Amazon S3`, you must configure the corresponding section too (you might continue to use the same configuration you set in the [adhoc full backup](#ad-hoc-full-backup)).
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
    * Set `name` to the name of the backup you want to restore (refer to [view backups](#view-backups) to view available backups). If the backup is stored in [Google Cloud Storage](https://cloud.google.com/storage/), [Ceph Object Storage](https://ceph.com/ceph-storage/object-storage/) or [Amazon S3](https://aws.amazon.com/s3/), you must configure the corresponding sections and make sure that the same configurations are applied as you perform the [full backup](#full-backup).
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+* Create a Kubernetes [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) containing the user and password that has the privilege to restore the database, and set `secretName` in the `values.yaml` file to the name of the created `Secret`(default to `backup-secret`, skip this if you have already created one in the [adhoc full backup](#ad-hoc-full-backup) section):
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
* Create a Kubernetes [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) containing the user and password (the user must have the database-related privileges to back up the data). Meanwhile, set `secretName` in the `values.yaml` file to the name of the created `Secret` (default to `backup-secret`; skip this if you have already created one when you perform [full backup](#full-backup)):
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    helm install pingcap/tidb-backup --namespace=<namespace> --name=<restore-name> -f values.yaml --version=<tidb-operator-version>
+    ```
+
+## Incremental backup
+
+Incremental backup leverages the [TiDB Binlog](https://www.pingcap.com/docs/dev/reference/tools/tidb-binlog/overview/) tool to collect binlog data from TiDB and provide real-time backup and replication to downstream platforms.
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
Incremental backup uses [TiDB Binlog](/dev/reference/tools/tidb-binlog/overview.md) to collect binlog data from TiDB and provide real-time backup and replication to downstream databases.
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+Incremental backup is disabled in the TiDB cluster by default. To create a TiDB cluster with incremental backup enabled or enable incremental backup in existing TiDB cluster, modify the `values.yaml` file:
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
Incremental backup is disabled in TiDB by default. To enable it, modify the `values.yaml` file:
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+* Set `binlog.pump.create` to `true`.
+* Set `binlog.drainer.create` to `true`.
+* Set `binlog.pump.storageClassName` and `binlog.drainer.storageClassName` to a proper `storageClass` available in your kubernetes cluster.
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
* Set `binlog.pump.storageClassName` and `binlog.drainer.storageClassName` to an available `storageClass` in your Kubernetes cluster.
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+* Set `binlog.drainer.destDBType` to your desired downstream, explained in detail below.
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
* Set `binlog.drainer.destDBType` to your desired downstream storage as needed, which is explained in details below.
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+Three types of downstream platforms available for incremental backup:
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
Incremental backup supports three types of downstream storage:
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+* PersistenceVolume: default downstream. You can consider configuring a large PV for `drainer` (the `binlog.drainer.storage` variable) in this case.
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
* PersistenceVolume: the default downstream storage. You can consider configuring a large PV for `drainer` (by modifying `binlog.drainer.storage`) in this case.
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+* MySQL compatible database: enabled by setting `binlog.drainer.destDBType` to `mysql`. You must configure the target address and credential in the `binlog.drainer.mysql` section too.
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
* MySQL compatible databases: enabled by setting `binlog.drainer.destDBType` to `mysql`. Meanwhile, you must configure the address and credential of the target database in `binlog.drainer.mysql`.
```

File: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+* Apache Kafka: enable by setting `binlog.drainer.destDBType` to `kafka`. You must configure the zookeeper address and Kafka address in the `binlog.drainer.kafka` section too.
```
Comment (yikeke) [inline dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md]:
```suggestion
* Apache Kafka: enabled by setting `binlog.drainer.destDBType` to `kafka`. Meanwhile, you must configure the zookeeper address and Kafka address of the target cluster in `binlog.drainer.kafka`.
```

**Comment by aylei** [comment]
@yikeke Thanks! I've addressed all the review comments, PTAL

**Pull request merged**